src/aof.c

```diff
@@ -1462,10 +1469,13 @@ int loadSingleAppendOnlyFile(char *filename) {
             ret = AOF_FAILED;
             goto cleanup;
         } else {
+            /* Restore the replication ID / offset from the RDB file. */
+            rsi_is_valid = rdbRestoreOffsetFromSaveInfo(&rsi, true);
             loadingAbsProgress(ftello(fp));
             last_progress_report_size = ftello(fp);
             if (old_style) serverLog(LL_NOTICE, "Reading the remaining AOF tail...");
         }
+        if (!rsi_is_valid && server.repl_backlog && listLength(server.replicas) == 0) freeReplicationBacklog();
```

Comment:
I was curious why we do this and I found the answer in a comment in the corresponding code for RDB load. Is it worth adding a similar comment here? A shorter one perhaps like this?

```suggestion
        /* If the AOF didn't contain replication info, it's not possible to
         * support partial resync, so we can free the backlog to save memory. */
        if (!rsi_is_valid && server.repl_backlog && listLength(server.replicas) == 0) freeReplicationBacklog();
```